go/vt/vterrors/vterrors.go

```diff
@@ -91,19 +91,35 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"sync"
 
 	"github.com/spf13/pflag"
 
 	vtrpcpb "vitess.io/vitess/go/vt/proto/vtrpc"
 )
 
-// logErrStacks controls whether or not printing errors includes the
+// logErrStacks controls whether printing errors includes the
 // embedded stack trace in the output.
 var logErrStacks bool
+var muLogErrStacks sync.Mutex
+
+func getLogErrStacks() bool {
+	muLogErrStacks.Lock()
+	defer muLogErrStacks.Unlock()
+	return logErrStacks
+}
+
+func setLogErrStacks(val bool) {
+	muLogErrStacks.Lock()
+	defer muLogErrStacks.Unlock()
+	logErrStacks = val
+}
 
 // RegisterFlags registers the command-line options that control vterror
 // behavior on the provided FlagSet.
 func RegisterFlags(fs *pflag.FlagSet) {
+	muLogErrStacks.Lock()
+	defer muLogErrStacks.Unlock()
 	fs.BoolVar(&logErrStacks, "log_err_stacks", false, "log stack traces for errors")
```

Comment:
just FYI this doesn't actually lock the `Set` the happens during flag parsing. if that's happening during tests you'll need to wrap the underlying bool in a custom `pflag.Value` implementation that does the locking during its `Set` call.

Reply:
To tie this back to my comment as well, we could do something like this:
```
diff --git a/go/vt/vterrors/vterrors.go b/go/vt/vterrors/vterrors.go
index 3b264b0110..78510d79ae 100644
--- a/go/vt/vterrors/vterrors.go
+++ b/go/vt/vterrors/vterrors.go
@@ -91,20 +91,49 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"sync/atomic"
 
 	"github.com/spf13/pflag"
 
 	vtrpcpb "vitess.io/vitess/go/vt/proto/vtrpc"
 )
 
+// atomicBoolValue is a wrapper for atomic.Bool that implements
+// pflag.Value.
+type atomicBoolValue struct {
+	atomic.Bool
+}
+
 // logErrStacks controls whether or not printing errors includes the
 // embedded stack trace in the output.
-var logErrStacks bool
+var logErrStacks *atomicBoolValue = &atomicBoolValue{}
+
+func (ab *atomicBoolValue) Type() string {
+	// This is the user facing type for the flag.
+	return "bool"
+}
+func (ab *atomicBoolValue) Set(val string) error {
+	switch val {
+	case "true":
+		ab.Store(true)
+	case "false":
+		ab.Store(false)
+	default:
+		return Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "invalid value for log_err_stacks: %s", val)
+	}
+	return nil
+}
+func (ab *atomicBoolValue) String() string {
+	if ab.Load() {
+		return "true"
+	}
+	return "false"
+}
 
 // RegisterFlags registers the command-line options that control vterror
 // behavior on the provided FlagSet.
 func RegisterFlags(fs *pflag.FlagSet) {
-	fs.BoolVar(&logErrStacks, "log_err_stacks", false, "log stack traces for errors")
+	fs.Var(logErrStacks, "log_err_stacks", "log stack traces for errors")
 }
 
 // New returns an error with the supplied message.
@@ -161,7 +190,7 @@ func (f *fundamental) Format(s fmt.State, verb rune) {
 	case 'v':
 		panicIfError(io.WriteString(s, "Code: "+f.code.String()+"\n"))
 		panicIfError(io.WriteString(s, f.msg+"\n"))
-		if logErrStacks {
+		if logErrStacks.Load() {
 			f.stack.Format(s, verb)
 		}
 		return
@@ -278,7 +307,7 @@ func (w *wrapping) Format(s fmt.State, verb rune) {
 	if rune('v') == verb {
 		panicIfError(fmt.Fprintf(s, "%v\n", w.Cause()))
 		panicIfError(io.WriteString(s, w.msg))
-		if logErrStacks {
+		if logErrStacks.Load() {
 			w.stack.Format(s, verb)
 		}
 		return

```